internal/querycoordv2/services.go

```diff
@@ -464,6 +465,107 @@ func (s *Server) GetSegmentInfo(ctx context.Context, req *querypb.GetSegmentInfo
 	}, nil
 }
 
+// RefreshCollection must be called after loading a collection. It looks for new segments that are not loaded yet and
+// tries to load them up. It returns when all segments of the given collection are loaded, or when error happens.
+// Note that a collection's loading progress always stays at 100% after a successful load and will not get updated
+// during RefreshCollection.
+func (s *Server) RefreshCollection(ctx context.Context, req *querypb.RefreshCollectionRequest) (*commonpb.Status, error) {
+	ctx, cancel := context.WithTimeout(ctx, Params.QueryCoordCfg.LoadTimeoutSeconds.GetAsDuration(time.Second))
+	defer cancel()
+
+	log := log.Ctx(ctx).With(
+		zap.Int64("collectionID", req.GetCollectionID()),
+	)
+	if s.status.Load() != commonpb.StateCode_Healthy {
+		msg := "failed to refresh collection"
+		log.Warn(msg, zap.Error(ErrNotHealthy))
+		metrics.QueryCoordReleaseCount.WithLabelValues(metrics.FailLabel).Inc()
+		return utils.WrapStatus(commonpb.ErrorCode_UnexpectedError, msg, ErrNotHealthy), nil
+	}
+
+	// Pull the latest target.
+	s.targetObserver.UpdateNextTarget(req.GetCollectionID())
```

Comment:
Can't update next target directly, as see target_observer.go, the line 91, 92, this may lead to update current target to a not ready next target